### ISSUE TITLE
Suggestions for improving Interpolation/Observation documentation

### ISSUE THREAD

**Issue by nikwit** [sxs-collaboration/spectre#3772]
When I tried to implement interpolation for the scalar wave system I found a few things missing from the docs that would have been quite useful to know. Here are a few suggestions to improve this

- The interpolation documentation is currently almost all contained in the docs for `InterpolationTarget`. However, this would feel more appropriate as a dev guide where it is also more easily found. 
- Mention the `InterpolateWithoutInterpComponent` action
- `observed_reduction_data_tags` is currently mentioned neither in the dev guides for observations nor interpolation though it is critical and gives cryptic error messages
- `InterpolationTargetTag` could use a protocol

**Comment by nilsvu** [comment]
Note on dev guides: They become more easily out of date because they are further away from the code. They are also not easily discoverable from the code. Therefore, I think it's very important to link to the dev guide from all relevant places in the code, e.g. with `\see`.

**Comment by knelli2** [comment]
@nikwit I agree, I think both the interpolation framework and the observation framework docs could use an update as I also found it confusing when trying to implement.

For observers, the dev guide [here](https://spectre-code.org/observers_infrastructure_dev_guide.html) actually does a pretty decent job at explaining how to do reductions and volume observations. I think it's lacking in a couple things
- like you said, explaining what `observed_reduction_data_tags` does and how important it is
- examples. Seeing a large wall of text is intimidating and very easy to get lost in. I think having concrete examples of components registering themselves and then contributing data for an observation would be very helpful

For interpolation I think your suggestions are good.
- Having all the docs on interpolation inside the `InterpolationTarget` does feel like an odd place to keep the broad overview of how interpolation works in spectre. @nilsvu brings up a good point about dev guides, but in this instance I think it would be very beneficial to have all this info in an easily accessible place right next to parallelization, observation, option parsing, etc...
- I'd be strongly for using a protocol for an `InterpolationTargetTag` because at the moment it's a lot of type aliases to remember.
- I've never used `InterpWithoutInterpComponent` but I know @markscheel will be working on clarifying/standardizing it
